tests/test_zarr_encryption_ipfs.py

```
@@ -0,0 +1,191 @@
+    print(
+        f"Took {total_time:.2f} seconds on average to load between the two loaded datasets"
+    )
+
+    # Test with bad encryption key
+    hamt1_read_bad = HAMT(
+        store=IPFSStore(),
+        root_node_id=hamt1_root,
+        read_only=True,
+        transformer=EncryptionFilter(encryption_key=get_random_bytes(32)),
+    )
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)